### ISSUE TITLE
In-Game: Large Blur Filter Effect not displaying correctly on macOS exports

### ISSUE THREAD

**Issue by ash0verride** [YoYoGames/GameMaker-Bugs#8205]
### Description
Issue when building project for macOS target where the Large Blur Filter Effect (https://manual.gamemaker.io/lts/en/The_Asset_Editors/Room_Properties/FX/All_Filter_Effect_Types.htm#:~:text=g_HueSaturation%20(Real)-,Large%20Blur,-%22_filter_large_blur%22) is not displaying correctly only on macOS exports

Effect is having issues when called in via an object that adjusts the runtime parameters over just adding into a room via the room editor layer effects options.

This issue is not experienced on other platforms

### Steps To Reproduce
1. Run the project on macOS.
2. Play through the game until a minigame starts up (triggered by the hazard icon).
3. The minigame when triggered will call and set the obj_minigame_blocker object to add the effect into the game.
4. The effect shown will appear to be a darken effect rather than just a blur. (displayed behind the minigame)

The project does not have this issue on other platforms.

 IDE v2024.8.1.171  Runtime v2024.8.1.218

### Which version of GameMaker are you reporting this issue for?
IDE v2024.8.1.171 Runtime v2024.8.1.218

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
macOS

[20499737-86ed-4303-85d0-c31a48121945](https://api.gamemaker.io/api/github/downloads/20499737-86ed-4303-85d0-c31a48121945)

**Comment by ash0verride** [comment]
[test_blur_issue.zip](https://github.com/user-attachments/files/17601377/test_blur_issue.zip)

uploaded a reduced down project still showing the issue

Windows (expected result)
![Image](https://github.com/user-attachments/assets/4b0aa3f8-9989-443f-9549-e0618b498cf4)

**Comment by ash0verride** [comment]
and mac (wrong result)
![Image](https://github.com/user-attachments/assets/4816dd58-c7c1-4e8a-a96d-c5d062e3bb8b)

**Comment by ash0verride** [comment]
Tested a build of main project on the macOS Sequoia - M3 iMac in office and issue didn't occur so may be machine specific

**Comment by JonnoRobson** [comment]
> Tested a build of main project on the macOS Sequoia - M3 iMac in office and issue didn't occur so may be machine specific

I've been able to reproduce the issue on a Mac with an Intel chipset. The Mac I was unable to reproduce the issue on was an Arm-based device so the problem must be related to specifically Intel-based devices. I'll investigated further to try to solve the issue.

**Comment by JonnoRobson** [comment]
The issue with the large blur filter on certain Mac devices was caused by using a floating point variable to control a for loop in the filter's fragment shader. On the devices displaying the issue the for loop would be unrolled incorrectly during shader compilation and the code contained within would only be executed 2 times rather than the expected 32 times. Replacing the variable with an integer fixes the issue so I've updated the shader to correct the problem.
